src/states/game.js

```diff
@@ -80,50 +62,110 @@ Game.prototype =
+			this.game.debug.body( this.__sprites.blackHole );
+		}
+	},
+
+	__createShip : function()
+	{
+		var ship = this.game.add.sprite(
+			100, 200, this.__assets.ship.name);
+		ship.anchor.set(0.5, 0.5);
+		this.game.physics.arcade.enable(ship);
+		ship.body.mass = this.__parameters.ship_mass;
+		ship.checkWorldBounds = true;
+		ship.events.onOutOfBounds.add(this.__shipOutOfBounds, true);
+		ship.body.velocity.setTo(60, 0);
+
+		return ship;
+	},
+
+	__createBlackHole : function()
+	{
+		var blackHole = this.game.add.sprite(
+			200, 300, this.__assets.blackHole.name);
+		blackHole.anchor.set(0.5, 0.5);
+		this.game.physics.arcade.enable(blackHole);
+		blackHole.body.mass = this.__parameters.blackHole_mass;
+
+		return blackHole;
+	},
+
     update : function()
     {
         var ship = this.__sprites.ship;
         var blackHole = this.__sprites.blackHole;
-        
+
         var totalGravity = (ship.body.mass * blackHole.body.mass) /
             Phaser.Math.distanceSq(ship.x, ship.y, blackHole.x, blackHole.y);
-        
+
         var angleBetweenShipBH = Phaser.Math.angleBetweenPoints(ship.position,
-            blackHole.position);
-        
-        ship.body.gravity.setTo(totalGravity * Math.cos(angleBetweenShipBH),
-                                totalGravity * Math.sin(angleBetweenShipBH));
+																blackHole.position);
+		ship.body.gravity.setTo(Math.cos(angleBetweenShipBH) * totalGravity,
+								Math.sin(angleBetweenShipBH) * totalGravity);
 
         ship.body.rotation = 270 + Phaser.Math.radToDeg(
             Phaser.Math.angleBetweenPoints(ship.body.velocity, this.__origin));
-        
+
         if (this.input.activePointer.leftButton.isDown) {
-            blackHole.position.setTo(
-                this.camera.x + this.input.mousePointer.position.x,
-                this.camera.y + this.input.mousePointer.position.y);
+            blackHole.position.setTo(this.camera.x + this.input.mousePointer.position.x,
+									 this.camera.y + this.input.mousePointer.position.y);
         }
 
-        if (ship.body.position.x >
-            this.__parameters.camera_cutoff_x &&
-            ship.body.velocity.x > 0) {
+		var shipPositionPastCutoff = ship.body.position.x > this.__parameters.camera_cutoff_x;
+		var shipMovingForwards = ship.body.velocity.x > 0;
+        if (shipPositionPastCutoff && shipMovingForwards) {
+			var shipXPositionDiff = ship.body.position.x - ship.body.prev.x;
+
             this.__non_player_objects.forEach(function(npo_sprite) {
-                npo_sprite.body.position.setTo(npo_sprite.body.position.x -
-                    (ship.body.position.x - ship.body.prev.x),
-                    npo_sprite.body.position.y);
-            });
-            this.offset_x = this.offset_x +
-                (ship.body.position.x - ship.body.prev.x);
-            ship.body.position.setTo(ship.body.prev.x, ship.body.position.y);
+                npo_sprite.body.position.x -= shipXPositionDiff;
```

Comment:
I guess not, but what's a bit weird is how it's works fine for a bit and then errors all of a sudden without the player doing anything particularly different. Still, an easy fix even if the cause is still a little hazy!